pkg/controller/releasemanager/incarnation.go

```diff
@@ -501,7 +520,12 @@ func (i *IncarnationCollection) deployed() []Incarnation {
 func (i *IncarnationCollection) releasable() []Incarnation {
 	r := []Incarnation{}
 	for _, i := range i.sorted() {
-		if i.status.State.Current == "canarying" {
+		switch i.status.State.Current {
+		case "canarying":
+			r = append(r, i)
+		case "canaried":
```

Comment:
shouldn't canaried and pendingrelease not be releasable? Isn't this used for the virtual service logic?

Reply:
You're right, my concern was the call to `releasable()` here https://github.com/Medium/picchu/blob/master/pkg/controller/releasemanager/syncer.go#L251

The Revision `alertRules` were being deleted when the state moved to canaried/pendingrelease (and recreated when the state moved to releasing) because these states were not included in `releasable()`. I'll do something else to fix this without side effects.